### PR TITLE
All files

### PR DESCRIPTION
**Pull request by Lukease**

### CODE REVIEW

File: vite-project/index.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
```
Comment (jpudysz) [inline vite-project/index.html:1]: .idea to gitnignore

File: vite-project/main.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
+const app = document.querySelector('.app')
+
+const createTicTacChoice = () => {
+    const ticTacBox = document.createElement('div')
```
Comment (jpudysz) [inline vite-project/main.js:4]: between assignments and instructions add one enter:
```
const ticTacBox = document.createElement('div')

ticTacBox.classList.add('app__box')
app.appendChild(ticTacBox)
```

File: vite-project/main.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    const circle = document.createElement('div')
+    circle.classList.add('circle')
+    boxRight.appendChild(circle)
+
+    return ticTacBox
+}
+
+const createTicTacArena = () => {
+    const ticTacArena = document.createElement('div')
+    ticTacArena.classList.add('app__arena')
```
Comment (jpudysz) [inline vite-project/main.js:40]: aa

File: vite-project/main.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    return ticTacArena
+}
+
+const ticTacBox = createTicTacChoice()
+const arena = createTicTacArena()
+
+let cubeArray = Array.from(new Array(9))
+
+cubeArray.forEach((object, index) => {
+    const newCube = document.createElement('div')
```
Comment (jpudysz) [inline vite-project/main.js:52]: aa

File: vite-project/main.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    newCube.classList.add('cube')
+    arena.appendChild(newCube)
+
+    newCube.addEventListener('click', () => {
+
```
Comment (jpudysz) [inline vite-project/main.js:57]: remove enter

File: vite-project/main.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        const choice = document.querySelector('.app__box--choice.choice-green').firstChild.classList.value
+
+        if (newCube.classList.contains('cube') && choice === 'circle' && !newCube.hasChildNodes()) {
+            const myChoice = document.createElement('div')
```
Comment (jpudysz) [inline vite-project/main.js:60]: aa

File: vite-project/main.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+            myChoice.classList.add('circle')
+            newCube.appendChild(myChoice)
+            arena.style.pointerEvents = 'none'
+            cubeArray[index] = 'O'
+        }
+        if (newCube.classList.contains('cube') && choice === 'cross' && !newCube.hasChildNodes()) {
+            const firstCross = document.createElement('div')
```
Comment (jpudysz) [inline vite-project/main.js:67]: aa

File: vite-project/main.js [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+            firstCross.classList.add('cross')
+            newCube.appendChild(firstCross)
+
+            const secondCross = document.createElement('div')
```
Comment (jpudysz) [inline vite-project/main.js:72]: aa

File: vite-project/main.js [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+            secondCross.classList.add('cross')
+            firstCross.appendChild(secondCross)
+            arena.style.pointerEvents = 'none'
+            cubeArray[index] = 'X'
+        }
+    })
+})
+
+cubeArray.fill('')
```
Comment (jpudysz) [inline vite-project/main.js]: you can do it there:
```
let cubeArray = Array
     .from(new Array(9))
     .map(() => '')
```

File: vite-project/main.js [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+const makeChoice = event => {
+
```
Comment (jpudysz) [inline vite-project/main.js]: remove enter

File: vite-project/main.js [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+        event.target.parentNode.classList.add('choice-green')
+    }
+}
+
+ticTacBox.addEventListener('click', makeChoice)
+
+const allCube = document.querySelectorAll('.cube')
+
+const play = () => {
+
```
Comment (jpudysz) [inline vite-project/main.js]: aa

File: vite-project/main.js [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+    ticTacBox.removeEventListener('click', makeChoice)
+
+    const choice = document.querySelector('.app__box--choice.choice-green').firstChild.classList.value
+
+    let indexes = cubeArray.map((field, index) => field ? null : index)
+    let filtered = indexes.filter(item => item !== null)
+    let randomNumber = Math.floor(Math.random() * filtered.length)
+    let randomNumberValue = filtered[randomNumber]
```
Comment (jpudysz) [inline vite-project/main.js]: don't use let if you're not mutating it

File: vite-project/main.js [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+
+    if (randomNumberValue === undefined) {
```
Comment (jpudysz) [inline vite-project/main.js:111]: `if(randomNumberValue)`

File: vite-project/main.js [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+        alert('Draw')
+    }
+
+    if (allCube[randomNumberValue].classList.contains('cube') && !allCube[randomNumberValue].hasChildNodes() && choice === 'circle') {
+        setTimeout(() => {
+            const firstCross = document.createElement('div')
```
Comment (jpudysz) [inline vite-project/main.js:118]: aa

File: vite-project/main.js [hunk 15]
```diff
@@ -0,0 +1,150 @@
...
+            firstCross.classList.add('cross')
+            allCube[randomNumberValue].appendChild(firstCross)
+
+            const secondCross = document.createElement('div')
```
Comment (jpudysz) [inline vite-project/main.js:123]: aa

File: vite-project/main.js [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+            secondCross.classList.add('cross')
+            firstCross.appendChild(secondCross)
+            arena.style.pointerEvents = 'auto'
+            cubeArray[randomNumberValue] = 'X'
+        }, 1000)
+    }
+
+    if (allCube[randomNumberValue].classList.contains('cube') && !allCube[randomNumberValue].hasChildNodes() && choice === 'cross') {
+        setTimeout(() => {
+            const myChoice = document.createElement('div')
```
Comment (jpudysz) [inline vite-project/main.js:134]: aa

File: vite-project/main.js [hunk 17]
```diff
@@ -0,0 +1,150 @@
...
+            myChoice.classList.add('circle')
+            allCube[randomNumberValue].appendChild(myChoice)
+            arena.style.pointerEvents = 'auto'
+            cubeArray[randomNumberValue] = 'O'
+        }, 1000)
+    }
+    const win = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]]
+
+    let xWin = win.find(result => result.every(index => cubeArray[index] === 'X'))
+    let oWin = win.find(result => result.every(index => cubeArray[index] === 'O'))
```
Comment (jpudysz) [inline vite-project/main.js]: use consts

File: vite-project/main.js [hunk 18]
```diff
@@ -0,0 +1,150 @@
...
+
+    if (xWin !== undefined) {
```
Comment (jpudysz) [inline vite-project/main.js]: `if (xWin)`

File: vite-project/main.js [hunk 19]
```diff
@@ -0,0 +1,150 @@
...
+        alert('X win the game')
+        window.location.reload()
+    }
+    if (oWin !== undefined) {
```
Comment (jpudysz) [inline vite-project/main.js:152]: enter between ifs

**Review by jpudysz — CHANGES_REQUESTED** [review]

File: vite-project/main.js (hunk 13 above)
Comment (Lukease) [reply vite-project/main.js:111]: if i do it, i will always have a alert 'draw', from this if

**Pull request merged**